WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -231,10 +231,19 @@ import WordPressComAnalytics
     public override func viewWillAppear(animated: Bool) {
         super.viewWillAppear(animated)
 
+        refreshTableViewIfNeeded()
+
         // Trigger layouts, if needed, to correct for any inherited layout changes, such as margins.
         refreshTableHeaderIfNeeded()
     }
 
+    func refreshTableViewIfNeeded() {
+        // For the saved post topic, posts could potentially be removed from the list on the details screen. This allows us to refresh the data when returning from that screen.
```

Comment:
Definitely, I found a few cases while testing where the update didn't work as expected, but I will do some digging in to WPTableViewHandler and see if I can find the underlying cause. (This workaround shouldn't be necessary)